src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    transient int size;
+    transient Node<T> first;
+    transient Node<T> last;
+
+    private static class Node<T> {
+        T item;
+        Node<T> next;
+        Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> before = last;
+        if (size == 1) {
+            before = first;
+        }
+        Node<T> node = new Node(before, value, null);
+        if (size == 0) {
+            first = node;
+        }
+        if (size >= 1) {
+            before.next = node;
+        }
+        last = node;
+        size++;
+
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isIndexGood(index);
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            Node<T> node = new Node<>(null, value, first);
+            first.prev = node;
+            first = node;
+            size++;
+            return;
+        } else {
+            Node<T> node = new Node<>(findNodeWithIndex(index).prev,
+                    value, findNodeWithIndex(index));
+            findNodeWithIndex(index).prev.next = node;
+            findNodeWithIndex(index).prev = node;
+            size++;
+            return;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexGood(index);
+        if (index == size) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+        return findNodeWithIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexGood(index);
+        if (index == size) {
+            throw new IndexOutOfBoundsException("Wring index");
+        }
+        T valueBeforeSetting = findNodeWithIndex(index).item;
+        findNodeWithIndex(index).item = value;
+        return valueBeforeSetting;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexGood(index);
+        if (index == size) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+        Node<T> node = findNodeWithIndex(index);
+        T item = node.item;
+        if (index == size - 1 && index == 0) {
+            first.item = null;
+            last.item = null;
+            size--;
```

Comment:
try simplifying this method)
hints:
let's move  `size--;` at the end of the method
the same with `node.item = null;`